sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/base_node.py

```diff
@@ -204,6 +204,40 @@ def name(self, value):
             value = value.lower()
         self._name = value
 
+    def get_predecessors(self) -> List["BaseNode"]:
```

Comment:
Just curious, how our customers will use this API?

Reply:
See `test_dsl_pipeline::test_get_predecessors` for use cases for this API.
For Det.Ch use case, Shrike will visit the pipeline nodes in topological order and check if predecessors of current node should use Det.Ch compute to determine if current node should use it.